packages/typescript-estree/src/convert.ts

```diff
@@ -2405,6 +2405,21 @@ export class Converter {
       // Binary Operations
 
       case SyntaxKind.BinaryExpression: {
+        if (
+          node.operatorToken.kind !== SyntaxKind.InKeyword &&
+          node.left.kind === SyntaxKind.PrivateIdentifier
+        ) {
+          this.#throwError(
+            node.left,
+            "Private identifiers are only allowed on the left-hand-side of an 'in' expression.",
```

Comment:
It might be more clear to say 

```suggestion
            "Private identifiers cannot appear on the right-hand-side of an 'in' expression.",
```

?

In an `in` expression, private identifiers are only allowed on the left side, but private identifiers aren't only allowed on the right hand side of an `in` expression 😅

Reply:
Coming back to this, I'd like to get this PR in today - so I'll apply the suggestion. If someone wants to advocate for a different message later on I'd be happy to take on the work to fix it up.